### DISCUSSION TITLE
Multiple root files

### DISCUSSION THREAD

**Discussion by felixlen** [James-Yu/LaTeX-Workshop#3493] [Q&A] [answered]
Dear all,

I have a use case where I input a file into two different root files, that is I have two files that contain a
`\begin{document} ... \include{subfile} ... \end{document}` pattern.

Now one of the two files is selected for building as is described in the Wiki section:

> Root directory check LaTeX Workshop iterates through all .tex files in the root folder of the workspace. The first one containing \begin{document} and which includes the file in the active editor is set as the root file. To avoid parsing all .tex files in the workspace, you can narrow the search by specifying [latex-workshop.latex.search.rootFiles.include](https://github.com/James-Yu/LaTeX-Workshop/discussions/new?category=q-a#latex-workshoplatexsearchrootFilesinclude) and/or [latex-workshop.latex.search.rootFiles.exclude](https://github.com/James-Yu/LaTeX-Workshop/discussions/new?category=q-a#latex-workshoplatexsearchrootFilesexclude).

I don't know if it is possible given the current design that uses exactly one root file to also have two root files. What I would like to achieve is that if I change something in `subfile.tex`, both root files will be built.

Many thanks for your feedback!
Felix

**Comment by James-Yu** [comment, marked as answer]
No, it’s not possible. And unlikely to be implemented. With multiple root files, you also need an interface to dynamically add and remove ones, think of ways to manage them, resolving file conflicts and more.

**Reply by felixlen** [reply]
Thanks a lot for the quick reply!
